skins/oasis/modules/OasisController.class.php

```diff
@@ -248,67 +239,15 @@ public function executeIndex($params) {
 		wfProfileOut(__METHOD__);
 	}
 
-	/**
-	 * Gets the URL and converts it to minified one if it points to single static file (JS or CSS)
-	 * If it's not recognized as static asset the original URL is returned
-	 *
-	 * @param $url string URL to be inspected
-	 * @return string
-	 */
-	private function minifySingleAsset( $url ) {
-		global $wgAllInOne;
-		if ( !empty( $wgAllInOne ) ) {
-			static $map;
-			if (empty($map)) {
-				$map = array(
-					array( $this->app->wg->ExtensionsPath, 'extensions/' ),
-					array( $this->app->wg->StylePath, 'skins/' ),
-					// $wgResourceBasePath = $wgCdnStylePath (there's no /resources in it)
-					array( $this->app->wg->ResourceBasePath . '/resources', 'resources/' ),
-				);
-			}
-
-			// BugId:38195 - don't minify already minified assets
-			if (strpos($url, '/__am/') !== false) {
-				return $url;
-			}
-
-			// don't minify already minified JS files
-			if (strpos($url, '.min.js') !== false) {
-				return $url;
-			}
-
-			foreach ($map as $item) {
-				list( $prefix, $replacement ) = $item;
-
-				// BugId: 38195 - wgExtensionPath / stylePath / ResourceBasePath do not end with a slash
-				// add one to remove double slashes in resulting URL
-				$prefix .= '/';
-
-				if (startsWith($url, $prefix)) {
-					$nurl = substr($url,strlen($prefix));
-					$matches = array();
-					if (preg_match("/^([^?]+)/",$nurl,$matches)) {
-						if (preg_match("/\\.(css|js)\$/i",$matches[1])) {
-							return $this->assetsManager->getOneCommonURL($replacement . $matches[1],true);
-						}
-					}
-				}
-			}
-		}
-		return $url;
-	}
-
 	// TODO: implement as a separate module?
 	private function loadJs() {
-		global $wgJsMimeType, $wgUser, $wgDevelEnvironment, $wgAllInOne;
+		global $wgJsMimeType, $wgUser;
 		wfProfileIn(__METHOD__);
 
 		$this->jsAtBottom = self::JsAtBottom();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)